Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -207,14 +209,67 @@ namespace PhysX
         *m_shapeConfig = Utils::CreateHeightfieldShapeConfiguration(GetEntityId());
     }
 
-    void EditorHeightfieldColliderComponent::RefreshHeightfield()
+    void EditorHeightfieldColliderComponent::RefreshHeightfield(const AZ::Aabb& dirtyRegion,
+        [[maybe_unused]] const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask* changeMask)
     {
-        ClearHeightfield();
-        InitHeightfieldShapeConfiguration();
+        AZ::Aabb heightfieldAabb = GetColliderShapeAabb();
+        AZ::Aabb requestRegion = dirtyRegion;
+
+        if (!requestRegion.IsValid())
+        {
+            requestRegion = heightfieldAabb;
+        }
+
+        // Early out if the updated region is outside of the heightfield Aabb
+        if (heightfieldAabb.Disjoint(requestRegion))
```

Comment:
Neat! I've only ever seen/used "if (!aabb.Overlaps())", I didn't realize we had a Disjoint() API. :)